inject.js

```diff
@@ -61,26 +64,27 @@ chrome.extension.sendMessage({}, function(response) {
         if (target.readyState === 0) {
           return;
         }
-        var speed = this.getSpeed();
-        this.speedIndicator.textContent = speed;
+        var speed = this.video.playbackRate;
+        this.speedIndicator.textContent = speed.toFixed(2);
         tc.settings.speed = speed;
-        chrome.storage.sync.set({'speed': speed});
+        if(speed !== 1.0) {
+          tc.settings.toggleSpeed = speed;
+          chrome.storage.sync.set({'speed': speed, 'toggleSpeed': speed});
+        } else {
+          chrome.storage.sync.set({'speed': speed});
+        }
```

Comment:
Would it make more sense to only store the reset speed value when reset action is invoked...?